keps/sig-cluster-lifecycle/20190227-artifact-generation.md

```diff
@@ -0,0 +1,197 @@
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Release Signoff Checklist
+
+**ACTION REQUIRED:** In order to merge code into a release, there must be an issue in [kubernetes/enhancements] referencing this KEP and targeting a release milestone **before [Enhancement Freeze](https://github.com/kubernetes/sig-release/tree/master/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Implementation of a standard, community supported and maintained process of generating and building rpm and deb packages.
+
+## Motivation
+
+Today, the tooling used to build kubernetes packages is separate from kubernetes itself. This can cause downstream divergence, and cannot be easily maintained by the kubernetes community as a whole. 
+
+The primary goal of this proposal is to simplify the lifecycle management of packaging by co-locating the sources and tooling for generating packages into the kubernetes/kubernetes repository. 
+
+### Goals
+
+To create a maintainable and configurable method of building kubernetes artifacts supported and maintained by the community.
+
+- Move packaging tools out of kubernetes/release and into kubernetes/kubernetes repository
+- Enable customization for downstream packaging 
+    - Including the ability to bring their own source code/binaries and changelogs 
+    - Stop hard-coding the binary download URL to dl.k8s.io
+- Enable community governance over packaging of the project.
+- Tightly couple the versioning of packing and the kubernetes/kubernetes release.
+- Ensure that the same packages that are run in CI are the ones that are released.
```

Comment:
not sure if this should be part of this KEP or the other KEP about publishing.

i fixed a bug yesterday related to CI packages:
https://github.com/kubernetes/test-infra/pull/11534

the CI packages are generated by bazel and pushed to GCS by a post-submit job.
to preserve the existing CI behavior and if bazel is no longed used for building debs/rpms, new prow jobs have to be created that exercise the new package generation tooling.

Reply:
I don't know that this would be true. Bazel could/should in fact call this to package the built binary. Meaning Bazel could continue to push to GCS afterwards. Also, I think that in the future per the other KEP mentioned we will start to push to CNCF so that part will be changing anyhow if I understand your question correctly.